core/src/test/java/io/wcm/testing/mock/aem/dam/MockGraniteAssetWrapperTest.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    @Test
+    public void testProperties() {
+        assertEquals("scott_reynolds.jpg", asset.getName());
+        assertEquals("/content/dam/sample/portraits/scott_reynolds.jpg", asset.getPath());
+        assertNotEquals(0, asset.hashCode());
+
+        if (context.resourceResolverType() == ResourceResolverType.JCR_OAK) {
+            assertNotNull(asset.getIdentifier());
+        }
+        else {
+            assertEquals("442d55b6-d534-4faf-9394-c9c20d095985", asset.getIdentifier());
+        }
+    }
+
+    @Test
+    public void testRenditions() {
+        List<Rendition> renditions = IteratorUtils.toList(asset.listRenditions());
+        assertEquals(4, renditions.size());
+        assertTrue(hasRendition(renditions, "cq5dam.thumbnail.48.48.png"));
+        assertEquals("original", asset.getRendition("original").getName());
+    }
+
+    private boolean hasRendition(List<Rendition> renditions, String renditionName) {
+        for (Rendition rendition : renditions) {
+            if (StringUtils.equals(rendition.getName(), renditionName)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Test
+    public void testRemoveNonExistingRendition() {
+        asset.removeRendition("non-existing");
+    }
+
+    @Test
+    public void testEquals() {
+        Asset asset1 = this.context.resourceResolver().getResource("/content/dam/sample/portraits/scott_reynolds.jpg").adaptTo(Asset.class);
+        Asset asset2 = this.context.resourceResolver().getResource("/content/dam/sample/portraits/scott_reynolds.jpg").adaptTo(Asset.class);
+
+        assertEquals(asset1, asset2);
+    }
+
+    private void doTestAddRemoveRendition(final String renditionName) {
+        InputStream is = new ByteArrayInputStream(BINARY_DATA);
+        Rendition rendition = asset.setRendition(renditionName, is, Map.of("jcr:mimeType", "application/octet-stream"));
+
+        assertNotNull(rendition);
+        assertNotNull(asset.getRendition(renditionName));
+        Resource resource = context.resourceResolver().getResource("/content/dam/sample/portraits/scott_reynolds.jpg/jcr:content/renditions/" + renditionName);
+        assertNotNull(resource);
+
+        Optional<DamEvent> damEvent = damEventHandler.getLastEvent();
+        assertTrue(damEvent.isPresent());
+        assertEquals(DamEvent.Type.RENDITION_UPDATED, damEvent.get().getType());
+        assertEquals(asset.getPath(), damEvent.get().getAssetPath());
+        assertEquals(rendition.getPath(), damEvent.get().getAdditionalInfo());
+
+        asset.removeRendition(renditionName);
+
+        assertNull(asset.getRendition(renditionName));
+        resource = context.resourceResolver().getResource("/content/dam/sample/portraits/scott_reynolds.jpg/jcr:content/renditions/" + renditionName);
+        assertNull(resource);
+
+        damEvent = damEventHandler.getLastEvent();
+        assertTrue(damEvent.isPresent());
```

Comment:
Simplifying `DamEventHandler` would result in the `isPresent()` and `get()` calls being removed, which saves some boilerplate

Reply:
existing code so out of scope for this rather small pr?